chore(client): update to webpack@5

marstamm opened the pull request:
closes https://github.com/camunda/camunda-modeler/issues/3060
<!--

Thanks for creating this pull request!

Please make sure to link the issue you are closing as "Closes #issueNr". 
This helps us to understand the context of this PR.

-->

nikku commented on client/karma.config.js:
```diff
@@ -105,6 +106,9 @@ module.exports = function(karma) {
       },
       plugins: [
+        new webpack.ProvidePlugin({
+          process: 'process/browser',
```
What do we need this for and can't we get rid of it with `DefinePlugin`?

marstamm replied:
Webpack@5 no longer provides nodeJS polyfills for browsers by default. In the tests, we use [`process.nextTick()` ](https://github.com/camunda/camunda-modeler/blob/58123a652018db7d73ce8274d98713eece3f6020/client/src/app/__tests__/AppParentSpec.js#L466), which now needs to be provided by this plugin.

The Define plugin is still needed so we can supply env variables

nikku replied:
Uhm... let's get rid of `process.nextTick?`

nikku commented on client/webpack.config.js:
```diff
@@ -37,23 +37,33 @@ const { LicenseWebpackPlugin } = require('license-webpack-plugin');
-    ]
+    ],
+    fallback: {
+      util: require.resolve('util/')
```
What do we need this for?

marstamm replied:
Similar to https://github.com/camunda/camunda-modeler/pull/3062#discussion_r940995410, `util` is used in the client (e.g. [util/inherits](https://github.com/camunda/camunda-modeler/blob/736fe6cef9d6857b0beef4ee0c0482cba85b298f/client/src/app/tabs/dmn/modeler/features/decision-table-keyboard/DecisionTableKeyboard.js#L11)) and needs to be provided by us in Webpack@5

nikku replied:
We should rather change to [`inherits-browser`](https://github.com/nikku/inherits-browser) which gets rid of the util import _or_ configure webpack to use the `browser` field over the `main` and `module` fields.

nikku commented on client/webpack.config.js:
```diff
@@ -127,7 +134,7 @@ module.exports = {
     writeToDisk: true
   },
-  devtool: DEV ? 'cheap-module-eval-source-map' : 'source-map'
+  devtool: DEV ? 'cheap-module-source-map' : 'source-map'
```
Why? :)

marstamm replied:
`cheap-module-eval-source-map` is not a valid devtool in webpack@5: https://webpack.js.org/configuration/devtool/

marstamm commented:
I applied the suggested changes and removed the `process.nextTick` usage and switched to `inherits-browser`.

I also removed a broken check for linux that was always false and would break the test if fixed (cf. [04909ec](https://github.com/camunda/camunda-modeler/pull/3062/commits/04909ec73bf339195aa41f18365e7b79de89c197), [broken CI run](https://github.com/camunda/camunda-modeler/actions/runs/2859447636))

nikku reviewed (approved):
Awesome stuff. Great showing of incremental improvement on multiple levels!
